compiler/msgs.nim

```diff
@@ -545,14 +554,14 @@ proc liMessage*(conf: ConfigRef; info: TLineInfo, msg: TMsgKind, arg: string,
       if msg == hintProcessing:
         msgWrite(conf, ".")
       else:
-        styledMsgWriteln(styleBright, loc, resetStyle, color, title, resetStyle, s, KindColor, kindmsg)
+        styledMsgWriteln(styleBright, loc, resetStyle, color, title, resetStyle, s, KindColor, kindmsg, UnitSep)
         if conf.hasHint(hintSource) and info != unknownLineInfo:
           conf.writeSurroundingSrc(info)
         if hintMsgOrigin in conf.mainPackageNotes:
```

Comment:
with `--hint:source` it seems to split at the wrong place, please check

Reply:
Good catch, fixed.